sympy/ntheory/generate.py

```diff
@@ -545,52 +551,56 @@ def nextprime(n, ith=1):
         primerange : Generate all primes in a given range
 
     """
-    n = int(n)
-    i = as_int(ith)
-    if i > 1:
-        pr = n
-        j = 1
-        while 1:
-            pr = nextprime(pr)
-            j += 1
-            if j > i:
-                break
-        return pr
-
-    if n < 2:
-        return 2
-    if n < 7:
-        return {2: 3, 3: 5, 4: 5, 5: 7, 6: 7}[n]
-    if n <= sieve._list[-2]:
-        l, u = sieve.search(n)
-        if l == u:
-            return sieve[u + 1]
+    @classmethod
+    def eval(cls, n, ith=1):
+        if not isinstance(n, Integer):
+            return None
+        n = int(n)
+        i = as_int(ith)
+        if i > 1:
+            pr = n
+            j = 1
+            while 1:
+                pr = nextprime(pr)
+                j += 1
+                if j > i:
+                    break
+            return pr
+
+        if n < 2:
+            return 2
+        if n < 7:
+            return {2: 3, 3: 5, 4: 5, 5: 7, 6: 7}[n]
+        if n <= sieve._list[-2]:
+            l, u = sieve.search(n)
+            if l == u:
+                return sieve[u + 1]
+            else:
+                return sieve[u]
+        nn = 6*(n//6)
+        if nn == n:
+            n += 1
+            if isprime(n):
+                return n
+            n += 4
+        elif n - nn == 5:
+            n += 2
+            if isprime(n):
+                return n
+            n += 4
         else:
-            return sieve[u]
-    nn = 6*(n//6)
-    if nn == n:
-        n += 1
-        if isprime(n):
-            return n
-        n += 4
-    elif n - nn == 5:
-        n += 2
-        if isprime(n):
-            return n
-        n += 4
-    else:
-        n = nn + 5
-    while 1:
-        if isprime(n):
-            return n
-        n += 2
-        if isprime(n):
-            return n
-        n += 4
+            n = nn + 5
+        while 1:
+            if isprime(n):
+                return n
+            n += 2
+            if isprime(n):
+                return n
+            n += 4
 
 
-def prevprime(n):
-    """ Return the largest prime smaller than n.
+class prevprime(Function):
+    """ Reprents the function which returns the largest prime smaller than n.
```

Comment:
Largest prime, which is smaller than n, so yeah, it should work for non-integers but the output should be an integer prime. For non-integer, the output shouldn't be unevaluated unless it's a symbol.

Reply:
There is a spelling error here, but also the text that it had before was better. 

Also, please read the documentation guide for docstrings https://docs.sympy.org/latest/documentation-style-guide.html.